hazelcast/src/main/java/com/hazelcast/cp/internal/operation/RaftReplicateOp.java

```diff
@@ -75,6 +72,33 @@ public final void run() {
         replicate(raftNode).whenCompleteAsync(this, CALLER_RUNS);
     }
 
+    private void onNullRaftNode(RaftService service) {
+        if (service.isRaftGroupDestroyed(groupId)) {
+            sendResponse(new CPGroupDestroyedException(groupId));
+        } else if (groupId.getId() != 0 && service.isDiscoveryCompleted() && service.isStartCompleted()) {
+            sendResponseAfterMetadataCheck(service);
+        } else {
+            sendResponse(new NotLeaderException(groupId, service.getLocalCPEndpoint(), null));
+        }
+    }
+
+    private void sendResponseAfterMetadataCheck(RaftService service) {
+        getNodeEngine().getExecutionService().execute(CP_SUBSYSTEM_EXECUTOR, () ->
```

Comment:
In my previous attempts, I've got the `IllegalThreadStateException` with `"Operation ... cannot be run in current thread: ... "`. So that's why the new executor is used.